pkgs/tools/security/step-ca/default.nix

```diff
@@ -2,30 +2,39 @@
 , lib
 , fetchFromGitHub
 , buildGoModule
+, coreutils
 , pcsclite
 , PCSC
 , pkg-config
 }:
 
 buildGoModule rec {
   pname = "step-ca";
-  version = "0.15.6";
+  version = "0.15.7-unstable";
 
   src = fetchFromGitHub {
     owner = "smallstep";
     repo = "certificates";
-    rev = "v${version}";
-    sha256 = "0n26692ph4q4cmrqammfazmx1k9p2bydwqc57q4hz5ni6jd31zbz";
+    rev = "4745be2309d7d146071831fc69c036c84b37f5f3";
+    sha256 = "gGJ96IhLhagOLbNJRz5A4uTSbJrXfuWp5g2Fvpu1pXQ=";
   };
 
-  vendorSha256 = "0w0phyqymcg2h2jjasxmkf4ryn4y1bqahcy94rs738cqr5ifyfbg";
+  vendorSha256 = "WDZhXxS7gqgmnZiafnJhw66EbhjPfL9AHvo3OtJI1D4=";
 
   nativeBuildInputs = [ pkg-config ];
 
   buildInputs =
     lib.optional stdenv.isLinux (lib.getDev pcsclite)
```

Comment:
pcsclite and PCSC are not strictly required to build step-ca from source. They are only required if you want to build it with YubiKey support. The default `make` rule will build a non-CGO version without YubiKey support, but we had a bug in our Makefile around this, which @maraino just fixed in smallstep/certificates#474.

Reply:
Thanks but as we run the tests I will keep it for now. I also don't know anything about our go infrastructure so I didn't manage to remove the support anyways. (Removing from `buildInputs` and disabling the tests wasn't enough)

Reply:
Do you have an older build? Just to be clear, we're releasing a new version of `step-ca` in the next 24-48 hours that will run `make test` without requiring `pcsclite`. 

So, I think this is really a matter of whether you want the nixOS package to have YubiKey support or not. The default build for `make` is a non-CGO (non-YubiKey supporting) `step-ca`.

Reply:
@tashian I found out that CGO is enabled by default on NixOS, I will push a fix soon.

Reply:
Done in https://github.com/NixOS/nixpkgs/pull/112322/commits/ec6555c278528705f922a7d517a62b6f105bac86. @midchildan I should probably add a `package` option now?

Reply:
@mohe2015 in the next release, if you compile with CGO, besides YubiKey support you will also have PKCS#11 support.

Reply:
@maraino Nice! This pull request is currently using the latest `master` commit and it seems like enabling CGO and YubiKey support using the option `securityTokenSupport` works (didn't test with a key but it compiles and runs). Is the name `securityTokenSupport` a sensible one (we could add additional options later)?

Reply:
`securityTokenSupport` won't be the right name, as we don't use those devices to 2FA, we use them to store keys, and do sign operations. @tashian what would be a proper name, `hsmSupport` (`hardwareSecurityModuleSupport`)?

Reply:
`securityModuleSupport` as the HSM can also be implemented in software (it should only be used for testing purposes)

Reply:
I like `hsmSupport` or `pkcs11Support`

Reply:
@mohe2015 

> @midchildan I should probably add a package option now?

I think that would be nice.

As to whether yubikey support should be enabled by default or not, that would depend on the number of users willing to use it. We can disable it for now and enable it later on if it becomes a popular feature, but I personally won't mind if we just enable it now because there's probably already a gazillion packages that depends on pcsclite anyways.

Reply:
@tashian @maraino  Any estimations how many users use the yubikey support?

@midchildan We could also enable it by default so users don't have to rebuild. It may be a larger attack surface though although I don't think that's actually relevant.

I personally don't really have an opinion on it but I don't use yubikey support myself.

Reply:
@mohe2015 In short, we'll see! We've had a lot of customers who need it for commercial use, mostly because they need FIPS 140 or other security compliance. Not sure how popular PKCS11 will be in open source. We got a lot of traction in the homelab community around our experimental YubiKey PIV support, when we posted our [Tiny CA tutorial](https://smallstep.com/blog/build-a-tiny-ca-with-raspberry-pi-yubikey/) in December.

Reply:
I enabled support for now so it's easier to get started as the pcsclite package is build already anyways.